connection.go

```diff
@@ -595,7 +592,9 @@ func (me *Connection) openChannel() (*Channel, error) {
 // this connection.
 func (me *Connection) closeChannel(ch *Channel, e *Error) {
 	ch.shutdown(e)
```

Comment:
You moved mutex from `func (me *Connection) releaseChannel(id uint16)` inside `func (me *Connection) closeChannel(ch *Channel, e *Error)`. And the reason was?